components/raftstore/src/store/worker/read.rs

```diff
@@ -1001,6 +1001,12 @@ where
         last_valid_ts: Timespec,
     ) -> Option<ReadResponse<E::Snapshot>> {
         let mut local_read_ctx = LocalReadContext::new(&mut self.snap_cache, read_id);
+        if snap_ctx.is_some() {
```

Comment:
The dependency of snap_ctx and range cache is too implicit and in the future it may change. 
We should have a more explicit way to address this issue. 

For example, we can have a method in snapshot : IsRegional() and if it's true, then it should not be cached at the first place.

Reply:
The requirement is that even when the in-memory-cache is enabled, for these not cached region, it can still keep current behavior. Only for cached regions, it will skip the snapshot cache.

Reply:
What you mean by `IsRegional`. If we want to use cache anyway, we may add a snapshot range method where RocksDB snapshot returns None meaning it's the whole range. We can compare it with the region of the delegate to see if we can reuse it.